Toggl.Ultrawave/Exceptions/ApiException.cs

```
@@ -34,26 +36,22 @@ public override string ToString()
                 + $"(Message: {message})";
 
         internal static string SerializeHeaders(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers)
-            => String.Join(", ", headers.Select(pair => $"'{pair.Key}': [{String.Join(", ", pair.Value.Select(v => $"'{v}'").ToArray())}]").ToArray());
-
-        public override string Message => ToString();
+            => Join(", ", headers.Select(pair => $"'{pair.Key}': [{Join(", ", pair.Value.Select(v => $"'{v}'").ToArray())}]").ToArray());
 
-        private string getLocalizedMessageFromResponse(IResponse response)
+        private static string getLocalizedMessageFromResponse(IResponse response)
         {
-            if (response.IsJson)
+            if (!response.IsJson)
+                return response.RawData;
+
+            try
             {
-                try
-                {
-                    var serializer = new JsonSerializer();
-                    var error = serializer.Deserialize<ResponseError>(response.RawData);
-                    return error.Message;
-                }
-                catch (DeserializationException<ResponseError>)
-                {
-                    return "";
-                }
+                var error = JsonConvert.DeserializeObject<ResponseError>(response.RawData);
+                return error?.Message ?? badJsonLocalisedError;
             }
-            return response.RawData;
+            catch (JsonSerializationException) { }
```

I was not here from the get-go... why do we swallow these exceptions here?
So the whole hell doesn't break loose when backend changes the API all of a sudden?